### PR TITLE
Introduce transactionLogDBName for DB2 on z/OS

### LINKED ISSUE #28507: DB2 for z/OS: transaction log manager should provide an option to change the default database
**Describe the use case that you want to enable:**
An option to change the default database when creating tables.
`CREATE TABLE (...) IN DATABASE database-name`

**Describe why this is important to you:**
Transaction log manager does not work with DB2 for z/OS if the default database, which is used for the JDBC driver, could not be used for the leases, partner and tranlog tables.

**Additional context**
The `databaseName` for "JDBC driver property" and `database-name` for "CREATE TABLE" are not the same.
- https://openliberty.io/docs/latest/reference/config/dataSource.html#properties
- https://www.ibm.com/docs/en/db2-for-zos/12?topic=statements-create-table

I am not a DBA, but I have understood this is a DB2 for z/OS related problem.

### PR DESCRIPTION
**Pull request by jonhawkes**
#build

### CODE REVIEW

File: dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -70,6 +70,9 @@ logging.for.heuristic.reporting.enabled.desc=Specifies whether the application s
 transaction.log.dbtablesuffix=Transaction log database table suffix
 transaction.log.dbtablesuffix.desc=When recovery logs are stored in an RDBMS table, this property allows the table name to be post-pended with a string to make it unique for this Server.
 
+transaction.log.db.name=Transaction log database name
+transaction.log.db.name.desc=Specifies the database name where recovery log tables are created.
```
Comment (dmuelle) [inline dev/com.ibm.ws.transaction/resources/OSGI-INF/l10n/metatype.properties]:
```suggestion
transaction.log.db.name.desc=Specifies the name of the database where recovery log tables are created.
```

**Review by dmuelle — COMMENTED** [review]
ID review

**Comment by jonhawkes** [comment]
#build

**Comment by jonhawkes** [comment]
#libby

**Comment by jonhawkes** [comment]
#libby
#build

**Comment by jonhawkes** [comment]
#libby
#build

**Comment by jonhawkes** [comment]
#libby
#build

**Review by djmatthews — DISMISSED** [review]

**Review by dmuelle — DISMISSED** [review]
LGTM, thanks

**Comment by jonhawkes** [comment]
#libby
#build

**Review by djmatthews — APPROVED** [review]

**Pull request merged**